Add `user defined types` section (#2)
tact-lang/spec: tact-spec.md

## New types

## User-defined types
User-defined types is a types which is defined by an used based on already existing types. 

### Product types

> Product types is a types that can have multiple fields of different types that exists as one entity. Besides of fields product types can also have functions that operates on type fields TODO.

Reviewer: ```suggestion
Product types are types that can have multiple fields of different types that exists as one entity. Product types have fields and methods — functions that operate on these fields.
```
